zmapsdk/input.py

```
@@ -1,243 +1,243 @@
+        Returns:
+            Path to the created allowlist file
+
+        Raises:
+            ZMapInputError: If a subnet is invalid or file can't be created
+        """
+        # Validate subnets
+        for subnet in subnets:
+            try:
+                ipaddress.ip_network(subnet)
+            except ValueError as e:
+                raise ZMapInputError(f"Invalid subnet in allowlist: {subnet} - {e!s}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)